core/Quasiquote.carp

```diff
@@ -0,0 +1,73 @@
+(doc Quasiquote "is a mechanism for quotation with code interspersed. See
+the documentation of [`quasiquote`](#quasiquote) or the [high level
+overview](https://github.com/carp-lang/Carp/blob/master/docs/Quasiquotation.md)
+for details.")
+(defmodule Quasiquote
+  (doc unquote "unquotes (i.e. evaluates) an expression `arg` inside a
+quasiquote. It will generate an error message when used outside a
+`quasiquote` form.
+
+Example:
+
+```
+(defdynamic x 1)
+
+(quasiquote (+ (unquote x) 1)) ; => (+ 1 1)
+```")
+  (defmacro unquote [arg]
+    (macro-error "unquotes need to be wrapped inside quasiquotes."))
+
+  (doc unquote "unquotes (i.e. evaluates) an expression `arg` inside a
```

Comment:
You probably mean `doc unquote-splicing` here.

Reply:
D'oh! Good catch, thank you!